core/beneficiary/l2_handler.go

```diff
@@ -0,0 +1,83 @@
+/*
+ * Copyright (C) 2021 The "MysteriumNetwork/node" Authors.
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ */
+
+package beneficiary
+
+import (
+	"fmt"
+
+	"github.com/ethereum/go-ethereum/common"
+	"github.com/mysteriumnetwork/node/identity"
+	"github.com/mysteriumnetwork/payments/crypto"
+)
+
+// l2Handler handles beneficiary in L2 chains.
+// Beneficiary is stored and retrieved from the given storage.
+type l2Handler struct {
+	set settler
+	st  storage
+
+	chainID               int64
+	hermesID              common.Address
+	registryAddress       common.Address
+	channelImplementation common.Address
+}
+
+const storageBucket = "beneficiary-bucket"
+
+func newL2Handler(chainID int64, hermesID, registryAddress, channelImplmentation common.Address, st storage, set settler) *l2Handler {
+	return &l2Handler{
+		set: set,
+		st:  st,
+
+		chainID:               chainID,
+		hermesID:              hermesID,
+		registryAddress:       registryAddress,
+		channelImplementation: channelImplmentation,
+	}
+}
+
+// SettleAndSaveBeneficiary settles beneficiary set to users own payments channel address.
+// The given beneficiary is saved to the database and later retrieved from there.
+func (b *l2Handler) SettleAndSaveBeneficiary(id identity.Identity, beneficiary common.Address) error {
+	addr, err := crypto.GenerateChannelAddress(
```

Comment:
I think it's best to base this off of https://github.com/mysteriumnetwork/node/pull/3027 as I've introduced a new component for both, channel address calculations as well as accessing hermes/registry/channel addresses.